

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Getting started &mdash; Blacktie 0.1_dev documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.1_dev',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Blacktie 0.1_dev documentation" href="index.html" />
    <link rel="next" title="Blacktie Auto-Generated Code Documentation" href="code.html" />
    <link rel="prev" title="Project Summary" href="project.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="code.html" title="Blacktie Auto-Generated Code Documentation"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="project.html" title="Project Summary"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Blacktie 0.1_dev documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Make sure that you have successfully installed the blacktie module before trying the activities below.</p>
</div>
<p>To test whether your installation was successful, open a new terminal session and type the following command.</p>
<div class="highlight-python"><pre>$ blacktie</pre>
</div>
<p>You should see the help text for blacktie and it should look something like this:</p>
<div class="highlight-none"><div class="highlight"><pre>usage: blacktie [-h] [--prog {tophat,cufflinks,cuffmerge,cuffdiff,all}]
                [--hide-logs] [--mode {analyze,dry_run,qsub_script}]
                config_file

This script reads options from a yaml formatted file and organizes the
execution of tophat/cufflinks runs for multiple condition sets.

positional arguments:
  config_file           Path to a yaml formatted config file containing setup
                        options for the runs.

optional arguments:
  -h, --help            show this help message and exit
  --prog {tophat,cufflinks,cuffmerge,cuffdiff,all}
                        Which program do you want to run? (default: tophat)
  --hide-logs           Make your log directories hidden to keep a tidy
                        &#39;looking&#39; base directory. (default: False)
  --mode {analyze,dry_run,qsub_script}
                        1) &#39;analyze&#39;: run the analysis pipeline. 2) &#39;dry_run&#39;:
                        walk through all steps that would be run and print out
                        the command lines; however, do not send the commands
                        to the system to be run. 3) &#39;qsub_script&#39;: generate
                        bash scripts suitable to be sent to a compute
                        cluster&#39;s SGE through the qsub command. (default:
                        analyze)
</pre></div>
</div>
<p>If this worked, great! Let&#8217;s move on to what all that means.</p>
<div class="section" id="the-configuration-file">
<h2>The configuration file<a class="headerlink" href="#the-configuration-file" title="Permalink to this headline">¶</a></h2>
<p>The configuration file is a <a class="reference external" href="http://en.wikipedia.org/wiki/YAML">YAML-based</a> document that is where we will store all of the complexity of the options, input and output files of the typical tophat/cufflinks workflow.  This way we have though about what we want to do with our RNA-seq data from start to finish before we actually start the analysis.  Also, this config file acts as a check on our poor memory.  If you get strange results you don&#8217;t have to worry about whether you entered the samples backwards since you can go back to this config file and see exactly what files and settings were used.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If you are running blacktie in <tt class="docutils literal"><span class="pre">analyze</span></tt> mode, you will have many more files created that document every step of the process where the output files are actually placed as well as central log files.</p>
</div>
<p>Here is a dummy example of a config file:</p>
<div class="highlight-yaml"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
 11
 12
 13
 14
 15
 16
 17
 18
 19
 20
 21
 22
 23
 24
 25
 26
 27
 28
 29
 30
 31
 32
 33
 34
 35
 36
 37
 38
 39
 40
 41
 42
 43
 44
 45
 46
 47
 48
 49
 50
 51
 52
 53
 54
 55
 56
 57
 58
 59
 60
 61
 62
 63
 64
 65
 66
 67
 68
 69
 70
 71
 72
 73
 74
 75
 76
 77
 78
 79
 80
 81
 82
 83
 84
 85
 86
 87
 88
 89
 90
 91
 92
 93
 94
 95
 96
 97
 98
 99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182</pre></div></td><td class="code"><div class="highlight"><pre><span class="c1"># The document starts after the &#39;---&#39;</span>

<span class="c1"># By the way: everything after a &#39;#&#39; on a line</span>
<span class="c1"># will be ignored by the program and acts as a</span>
<span class="c1"># comment or note to explain things.</span>

<span class="nn">---</span>
<span class="c1"># run_options is a dictionary that contains variables that will be needed for</span>
<span class="c1"># many or all stages of the run</span>
<span class="l-Scalar-Plain">run_options</span><span class="p-Indicator">:</span>
  <span class="l-Scalar-Plain">base_dir</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/project/base_dir</span>
  <span class="l-Scalar-Plain">run_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">False</span>         <span class="c1">#  name your run: if false; uses current date/time for uniqe run_id everytime</span>
  <span class="l-Scalar-Plain">bowtie_indexes_dir</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/bowtie2_indexes</span>
  <span class="l-Scalar-Plain">email_info</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">sender</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_me@gmail.com</span>           <span class="c1"># for now only sending from gmail accounts is supported (will change soon)</span>
    <span class="l-Scalar-Plain">to</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">to_you@email.com</span>
    <span class="l-Scalar-Plain">li</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/file/containing/base64_encoded/login_info</span>      <span class="c1"># base64_encoded pswrd for from_me@email.com</span>



<span class="c1"># `tophat_options`:</span>
<span class="c1"># -----------------</span>
<span class="c1"># This is a dictionary that contains variables needed for all the tophat runs.</span>
<span class="c1"># The names of the key:value combinations are taken directly from the tophat</span>
<span class="c1"># option names but have the leading &#39;-&#39; removed.</span>

<span class="c1"># -o becomes o; --library-type becomes library-type</span>

<span class="c1"># **This is true for the cufflinks, cuffmerge, cuffdiff option dictionaries.**</span>

<span class="c1"># `from_conditions`:</span>
<span class="c1"># ------------------</span>
<span class="c1"># This is a special value that tells blacktie that you don&#39;t want to name a single</span>
<span class="c1"># value for this option but would rather set the value individually for each of</span>
<span class="c1"># your samples/conditions.  If you set the `o` value here:</span>

<span class="c1">#    **all of your different sample results would</span>
<span class="c1">#      be written to the same output directory and</span>
<span class="c1">#      each would overwrite the next!**</span>
<span class="c1"># Hence: from_conditions</span>

<span class="c1"># However if you made all of your libraries the same way, things like `r` and</span>
<span class="c1"># `mate-std-dev` can be set here to avoid writing the same values over and over</span>
<span class="c1"># and perhaps making a mistake or two.</span>

<span class="c1"># `positional_args`:</span>
<span class="c1"># ------------------</span>
<span class="c1"># This is a dictionary inside of the `tophat_options` dictionary.</span>
<span class="c1"># It is where you put the arguments to tophat that do not have &#39;flags&#39; to make</span>
<span class="c1"># their identity explicit like `-o path/to/output_dir` or `--library-type fr-unstranded`</span>

<span class="c1"># For tophat, these values are</span>
<span class="c1">#     [1] the bowtie index name</span>
<span class="c1">#     [2] the fastq files containing the left_reads</span>
<span class="c1">#     [3] the fastq files containing the right_reads</span>

<span class="c1"># They will be different for cufflinks, cuffmerge, cuffdiff so consult the</span>
<span class="c1"># respective help text or manuals, but you should be fine if you just use what</span>
<span class="c1"># I have set up in this file already.</span>

<span class="l-Scalar-Plain">tophat_options</span><span class="p-Indicator">:</span>
  <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">library-type</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">fr-unstranded</span>
  <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">6</span>
  <span class="l-Scalar-Plain">r</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">125</span>
  <span class="l-Scalar-Plain">mate-std-dev</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">25</span>
  <span class="l-Scalar-Plain">G</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">no-coverage-search</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
  <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>

<span class="l-Scalar-Plain">cufflinks_options</span><span class="p-Indicator">:</span>
  <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">7</span>
  <span class="l-Scalar-Plain">GTF-guide</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">3-overhang-tolerance</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">5000</span>
  <span class="l-Scalar-Plain">frag-bias-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span> <span class="c1"># if not False; path to genome fasta</span>
  <span class="l-Scalar-Plain">multi-read-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
  <span class="l-Scalar-Plain">upper-quartile-norm</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
  <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">accepted_hits</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>

<span class="l-Scalar-Plain">cuffmerge_options</span><span class="p-Indicator">:</span>
  <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span> <span class="c1"># output directory</span>
  <span class="l-Scalar-Plain">ref-gtf</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">6</span>
  <span class="l-Scalar-Plain">ref-sequence</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">assembly_list</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span> <span class="c1"># file with path to cufflinks gtf files to be merged</span>

<span class="l-Scalar-Plain">cuffdiff_options</span><span class="p-Indicator">:</span>
  <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">labels</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">6</span>
  <span class="l-Scalar-Plain">time-series</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
  <span class="l-Scalar-Plain">upper-quartile-norm</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
  <span class="l-Scalar-Plain">frag-bias-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
  <span class="l-Scalar-Plain">multi-read-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
  <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">transcripts_gtf</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">sample_bams</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>


<span class="c1"># `condition_queue`:</span>
<span class="c1"># ------------------</span>
<span class="c1"># This is a list of info related to each sample/condition contained in your RNA-sequence</span>
<span class="c1"># experiment(s)</span>

<span class="c1"># `name`: the name of this condition and suffix of the `output_dir` for each</span>
<span class="c1">#         program.  Usually something like a time-point ID or treatment type.</span>
<span class="c1">#         Should be as short as possible while still being a useful label.</span>

<span class="c1"># `group_id`: this is how you group different experiments to be included in a</span>
<span class="c1">#             single cuffmerge/cuffdiff program call.  All conditions in a time</span>
<span class="c1">#             series should share the same `group_id` and be placed in</span>
<span class="c1">#             `condition_queue` in the order that you want them to be sent to</span>
<span class="c1">#             cuffdiff.</span>

<span class="c1"># `left_reads`: a list of the paths to fastq files containing left reads for</span>
<span class="c1">#               each condition.</span>

<span class="c1"># `right_reads`: list of fastqs containing the right mates for the fastqs in</span>
<span class="c1">#                `left_reads`.</span>
<span class="c1">#                 **NOTE** right mate file must be in same order as provided to `left_reads`</span>

<span class="l-Scalar-Plain">condition_queue</span><span class="p-Indicator">:</span>
  <span class="p-Indicator">-</span>
    <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp1_control</span>
    <span class="l-Scalar-Plain">group_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
    <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep1.left_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep2.left_reads.fastq</span>
    <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep1.right_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep2.right_reads.fastq</span>
    <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
    <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
    <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>

  <span class="p-Indicator">-</span>
    <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp1_treatment</span>
    <span class="l-Scalar-Plain">group_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
    <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep1.left_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep2.left_reads.fastq</span>
    <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep1.right_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep2.right_reads.fastq</span>
    <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
    <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
    <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>

  <span class="p-Indicator">-</span>
    <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp2_control</span>
    <span class="l-Scalar-Plain">group_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">1</span>
    <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep1.left_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep2.left_reads.fastq</span>
    <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep1.right_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep2.right_reads.fastq</span>
    <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
    <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
    <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>

  <span class="p-Indicator">-</span>
    <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp2_treatment</span>
    <span class="l-Scalar-Plain">group_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">1</span>
    <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep1.left_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep2.left_reads.fastq</span>
    <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep1.right_reads.fastq</span>
      <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep2.right_reads.fastq</span>
    <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
    <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
    <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>


<span class="nn">...</span>
</pre></div>
</td></tr></table></div>
</div>
<div class="section" id="using-e-mail-notifications">
<h2>Using e-mail notifications<a class="headerlink" href="#using-e-mail-notifications" title="Permalink to this headline">¶</a></h2>
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ipsum risus, scelerisque vitae consequat sit amet, placerat vitae turpis. Aliquam cursus justo vitae quam convallis vel auctor dui tristique. Nam justo nisl, pretium a scelerisque dictum, tincidunt sit amet diam. Donec neque nisi, ornare quis varius nec, scelerisque nec tellus. Maecenas neque purus, lobortis ac malesuada ut, vestibulum non dui. Proin tempor dolor est, quis facilisis orci. Maecenas varius dolor nec urna pellentesque adipiscing. Quisque mi lorem, fringilla non elementum quis, cursus sed nibh. Phasellus pellentesque turpis non enim interdum eget molestie purus interdum. Nunc id nunc justo, sit amet euismod velit.</p>
</div>
<div class="section" id="using-different-modes">
<h2>Using different modes<a class="headerlink" href="#using-different-modes" title="Permalink to this headline">¶</a></h2>
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ipsum risus, scelerisque vitae consequat sit amet, placerat vitae turpis. Aliquam cursus justo vitae quam convallis vel auctor dui tristique. Nam justo nisl, pretium a scelerisque dictum, tincidunt sit amet diam. Donec neque nisi, ornare quis varius nec, scelerisque nec tellus. Maecenas neque purus, lobortis ac malesuada ut, vestibulum non dui. Proin tempor dolor est, quis facilisis orci. Maecenas varius dolor nec urna pellentesque adipiscing. Quisque mi lorem, fringilla non elementum quis, cursus sed nibh. Phasellus pellentesque turpis non enim interdum eget molestie purus interdum. Nunc id nunc justo, sit amet euismod velit.</p>
</div>
</div>
<div class="section" id="tutorial">
<h1>Tutorial<a class="headerlink" href="#tutorial" title="Permalink to this headline">¶</a></h1>
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ipsum risus, scelerisque vitae consequat sit amet, placerat vitae turpis. Aliquam cursus justo vitae quam convallis vel auctor dui tristique. Nam justo nisl, pretium a scelerisque dictum, tincidunt sit amet diam. Donec neque nisi, ornare quis varius nec, scelerisque nec tellus. Maecenas neque purus, lobortis ac malesuada ut, vestibulum non dui. Proin tempor dolor est, quis facilisis orci. Maecenas varius dolor nec urna pellentesque adipiscing. Quisque mi lorem, fringilla non elementum quis, cursus sed nibh. Phasellus pellentesque turpis non enim interdum eget molestie purus interdum. Nunc id nunc justo, sit amet euismod velit.</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Getting started</a><ul>
<li><a class="reference internal" href="#the-configuration-file">The configuration file</a></li>
<li><a class="reference internal" href="#using-e-mail-notifications">Using e-mail notifications</a></li>
<li><a class="reference internal" href="#using-different-modes">Using different modes</a></li>
</ul>
</li>
<li><a class="reference internal" href="#tutorial">Tutorial</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="project.html"
                        title="previous chapter">Project Summary</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="code.html"
                        title="next chapter">Blacktie Auto-Generated Code Documentation</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/tutorial.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="code.html" title="Blacktie Auto-Generated Code Documentation"
             >next</a> |</li>
        <li class="right" >
          <a href="project.html" title="Project Summary"
             >previous</a> |</li>
        <li><a href="index.html">Blacktie 0.1_dev documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Augustine Dunn.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>