<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Getting started &mdash; Blacktie 0.2.1.2 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.2.1.2',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Blacktie 0.2.1.2 documentation" href="index.html" />
    <link rel="next" title="Blacktie Auto-Generated Code Documentation" href="code.html" />
    <link rel="prev" title="Installation" href="install.html" />
 
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-39589366-2', 'github.io');
  ga('send', 'pageview');

</script>

  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="code.html" title="Blacktie Auto-Generated Code Documentation"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="install.html" title="Installation"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">Blacktie 0.2.1.2 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Make sure that you have successfully installed the blacktie module before trying the activities below.</p>
</div>
<p>To test whether your installation was successful, open a new terminal session and type the following command.</p>
<div class="highlight-python"><pre>$ blacktie</pre>
</div>
<p>You should see the help text for blacktie and it should look something like this:</p>
<div class="highlight-none"><div class="highlight"><pre>$ blacktie

usage: blacktie [-h] [--version]
    [--prog {tophat,cufflinks,cuffmerge,cuffdiff,all}]
    [--hide-logs] [--no-email]
    [--mode {analyze,dry_run,qsub_script}]
    config_file

This script reads options from a yaml formatted file and organizes the
execution of tophat/cufflinks runs for multiple condition sets.

positional arguments:
    config_file           Path to a yaml formatted config file containing setup
        options for the runs.

optional arguments:
    -h, --help            show this help message and exit
    --version             Print version number.
    --prog {tophat,cufflinks,cuffmerge,cuffdiff,all}
        Which program do you want to run? (default: tophat)
    --hide-logs           Make your log directories hidden to keep a tidy
        &#39;looking&#39; base directory. (default: False)
    --no-email            Don&#39;t send email notifications. (default: False)
    --mode {analyze,dry_run,qsub_script}
        1) &#39;analyze&#39;: run the analysis pipeline. 2) &#39;dry_run&#39;:
        walk through all steps that would be run and print out
        the command lines; however, do not send the commands
        to the system to be run. 3) &#39;qsub_script&#39;: generate
        bash scripts suitable to be sent to a compute
        cluster&#39;s SGE through the qsub command. (default:
        analyze)
</pre></div>
</div>
<p>If this worked, great! Let&#8217;s move on to what all that means.</p>
<div class="section" id="the-prog-option">
<h2>The &#8211;prog option<a class="headerlink" href="#the-prog-option" title="Permalink to this headline">¶</a></h2>
<p>This tells <tt class="docutils literal"><span class="pre">blacktie</span></tt> which part of the pipeline you would like to run.  Any part can be run individually as long as the correct files exist.  You can also run the whole thing from <tt class="docutils literal"><span class="pre">tophat</span></tt> to <tt class="docutils literal"><span class="pre">cuffdiff</span></tt> in one fell swoop if you like!</p>
</div>
<div class="section" id="the-hide-logs-option">
<h2>The &#8211;hide-logs option<a class="headerlink" href="#the-hide-logs-option" title="Permalink to this headline">¶</a></h2>
<p>This names your log files so that they are hidden in &#8220;*nix&#8221; systems.</p>
</div>
<div class="section" id="the-modes-option">
<h2>The &#8211;modes option<a class="headerlink" href="#the-modes-option" title="Permalink to this headline">¶</a></h2>
<p><tt class="docutils literal"><span class="pre">blacktie</span></tt> can run in three modes.  The first, <tt class="docutils literal"><span class="pre">analyze</span></tt>, actually runs the pipeline and does the analyses.  However, it can be useful to simply view what WOULD be done to make sure that <tt class="docutils literal"><span class="pre">`blacktie</span></tt> is producing command line calls that match what you expected.  For this, use the <tt class="docutils literal"><span class="pre">dry_run</span></tt> mode.</p>
<p>Further, if you are working on a compute cluster running something like a &#8220;Sun Grid Engine&#8221; (<a class="reference external" href="http://en.wikipedia.org/wiki/Oracle_Grid_Engine">SGE</a>) to which you must <a class="reference external" href="https://wikis.utexas.edu/display/CCBB/sge-tutorial">submit jobs</a> using <tt class="docutils literal"><span class="pre">qsub</span></tt>, it may not be a good idea to submit a job running all of <tt class="docutils literal"><span class="pre">blacktie</span></tt> as a single <tt class="docutils literal"><span class="pre">qsub</span></tt> job.  For this it can be helpful to have <tt class="docutils literal"><span class="pre">blacktie</span></tt> write all of your <tt class="docutils literal"><span class="pre">qsub</span></tt> scripts for you based on a template. Each bash script represents a single program call to the tophat/cufflinks suite.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">A starter template for SGE submission can be found here: <a class="reference download internal" href="_downloads/qsub.template"><tt class="xref download docutils literal"><span class="pre">blacktie/examples/qsub.template</span></tt></a>.  You will want to become familiar with how <a class="reference external" href="http://www.makotemplates.org/">Mako</a> processes templates if you plan to customize this much.</p>
</div>
<p>Here is what the starter template looks like:</p>
<div class="highlight-bash"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50</pre></div></td><td class="code"><div class="highlight"><pre><span class="c">#!/bin/bash</span>
<span class="c">#$ -S /bin/bash                                             # Use a real BASH shell on the worker node</span>
<span class="c">#$ -q ${queues}                                             # What queues do you want to submit to</span>
<span class="c">#$ -M ${email_addy}                                         # Send email updates to this address</span>
<span class="c">#$ -m beas                                                  # When to send an email update</span>
<span class="c">#$ -e /data/users/dunnw/logs/${call_id}.e                   # Write standard error to this file</span>
<span class="c">#$ -o /data/users/dunnw/logs/${call_id}.o                   # Write standard out to this file</span>
<span class="c">#$ -N ${job_name}                                           # Name my job this</span>
<span class="c">#$ -R y                                                     # Reserve cores for me until there are the number I asked for</span>
<span class="c">#$ -pe openmp ${core_range}                                 # Use openmp for multiprocessor use and give me core_range cores</span>

<span class="nv">LD_LIBRARY_PATH</span><span class="o">=</span><span class="s2">&quot;${ld_library_path}$${}{LD_LIBRARY_PATH}&quot;</span>   <span class="c"># Make sure worker&#39;s LD_LIBRARY_PATH contains ld_library_path</span>


<span class="c"># HPC clusters frequently use a module system to provide system wide access to </span>
<span class="c"># certain programs.  The following makes sure that the tools needed are loaded</span>
<span class="c"># for **MY** cluster. You will need alter this to make sure your cluster is set up</span>
<span class="c"># based on its system.</span>

module load bowtie2/2.0.2
module load tophat/2.0.6
module load cufflinks/2.0.2
module load samtools/0.1.18


<span class="c"># basic staging stuff</span>
<span class="nv">DATAHOME</span><span class="o">=</span><span class="s2">&quot;${datahome}&quot;</span>
<span class="nv">MYSCRATCH</span><span class="o">=</span><span class="s2">&quot;/scratch/$${}{USER}&quot;</span>


mkdir -p <span class="nv">$MYSCRATCH</span>
<span class="nb">cd</span> <span class="nv">$MYSCRATCH</span>


<span class="c"># Remind me what will be done</span>
<span class="nb">echo</span> <span class="s1">&#39;&#39;</span>
<span class="nb">echo</span> <span class="s2">&quot;${cmd_str}&quot;</span>
<span class="nb">echo</span> <span class="s1">&#39;&#39;</span>

<span class="c"># Run my job</span>
<span class="k">${</span><span class="nv">cmd_str</span><span class="k">}</span>


<span class="c"># Pack up results and send it home to log-in node</span>
tar -zcvf <span class="k">${</span><span class="nv">call_id</span><span class="k">}</span>.tar.gz <span class="k">${</span><span class="nv">out_dir</span><span class="k">}</span>
cp <span class="k">${</span><span class="nv">call_id</span><span class="k">}</span>.tar.gz <span class="nv">$$</span><span class="o">{}{</span>DATAHOME<span class="o">}</span>/

<span class="c"># Back into the shadows</span>
<span class="nb">cd</span> <span class="nv">$HOME</span>
rm -rf <span class="nv">$MYSCRATCH</span>
</pre></div>
</td></tr></table></div>
</div>
<div class="section" id="the-configuration-file">
<h2>The configuration file<a class="headerlink" href="#the-configuration-file" title="Permalink to this headline">¶</a></h2>
<p>The configuration file is a <a class="reference external" href="http://en.wikipedia.org/wiki/YAML">YAML-based</a> document that is where we will store all of the complexity of the options, input and output files of the typical tophat/cufflinks workflow.  This way we have though about what we want to do with our RNA-seq data from start to finish before we actually start the analysis.  Also, this config file acts as a check on our poor memory.  If you get strange results you don&#8217;t have to worry about whether you entered the samples backwards since you can go back to this config file and see exactly what files and settings were used.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If you are running blacktie in <tt class="docutils literal"><span class="pre">analyze</span></tt> mode, you will have many more files created that document every step of the process where the output files are actually placed as well as central log files.</p>
</div>
<p>Here is a dummy example of a config file:</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">A copy of this file can be found here: <a class="reference download internal" href="_downloads/blacktie_config_example.yaml"><tt class="xref download docutils literal"><span class="pre">blacktie/examples/blacktie_config_example.yaml</span></tt></a></p>
</div>
<div class="highlight-yaml"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre>  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
 11
 12
 13
 14
 15
 16
 17
 18
 19
 20
 21
 22
 23
 24
 25
 26
 27
 28
 29
 30
 31
 32
 33
 34
 35
 36
 37
 38
 39
 40
 41
 42
 43
 44
 45
 46
 47
 48
 49
 50
 51
 52
 53
 54
 55
 56
 57
 58
 59
 60
 61
 62
 63
 64
 65
 66
 67
 68
 69
 70
 71
 72
 73
 74
 75
 76
 77
 78
 79
 80
 81
 82
 83
 84
 85
 86
 87
 88
 89
 90
 91
 92
 93
 94
 95
 96
 97
 98
 99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224</pre></div></td><td class="code"><div class="highlight"><pre><span class="c1"># The document starts after the &#39;---&#39;</span>

<span class="c1"># By the way: everything after a &#39;#&#39; on a line</span>
<span class="c1"># will be ignored by the program and acts as a</span>
<span class="c1"># comment or note to explain things.</span>

<span class="nn">---</span>
<span class="c1"># run_options is a dictionary that contains variables that will be needed for</span>
<span class="c1"># many or all stages of the run</span>
<span class="l-Scalar-Plain">run_options</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">base_dir</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/project/base_dir</span>
    <span class="l-Scalar-Plain">run_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">False</span>            <span class="c1">#  name your run: if false; uses current date/time for uniqe run_id everytime</span>
    <span class="l-Scalar-Plain">bowtie_indexes_dir</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/bowtie2_indexes</span> 
    <span class="l-Scalar-Plain">email_info</span><span class="p-Indicator">:</span>
        <span class="l-Scalar-Plain">sender</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_me@gmail.com</span>           
        <span class="l-Scalar-Plain">to</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">to_you@email.com</span>
        <span class="l-Scalar-Plain">li</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/file/containing/base64_encoded/login_info</span>      <span class="c1"># base64_encoded pswrd for from_me@email.com</span>
    <span class="l-Scalar-Plain">custom_smtp</span><span class="p-Indicator">:</span> 
        <span class="l-Scalar-Plain">host</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">smtp.gmail.com</span>   <span class="c1"># or what ever your email smtp server is</span>
        <span class="l-Scalar-Plain">port</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">587</span>              <span class="c1"># or which ever port your smtp server uses</span>



<span class="c1"># `tophat_options`:</span>
<span class="c1"># -----------------</span>
<span class="c1"># This is a dictionary that contains variables needed for all the tophat runs.</span>
<span class="c1"># The names of the key:value combinations are taken directly from the tophat</span>
<span class="c1"># option names but have the leading &#39;-&#39; removed.</span>

<span class="c1"># -o becomes o; --library-type becomes library-type</span>

<span class="c1"># **This is true for the cufflinks, cuffmerge, cuffdiff option dictionaries.** </span>

<span class="c1"># `from_conditions`:</span>
<span class="c1"># ------------------</span>
<span class="c1"># This is a special value that tells blacktie that you don&#39;t want to name a single</span>
<span class="c1"># value for this option but would rather set the value individually for each of</span>
<span class="c1"># your samples/conditions.  If you set the `o` value here: </span>

<span class="c1">#    **all of your different sample results would</span>
<span class="c1">#      be written to the same output directory and</span>
<span class="c1">#      each would overwrite the next!**</span>
<span class="c1"># Hence: from_conditions</span>

<span class="c1"># However if you made all of your libraries the same way, things like `r` and</span>
<span class="c1"># `mate-std-dev` can be set here to avoid writing the same values over and over</span>
<span class="c1"># and perhaps making a mistake or two.</span>

<span class="c1"># `positional_args`:</span>
<span class="c1"># ------------------</span>
<span class="c1"># This is a dictionary inside of the `tophat_options` dictionary.</span>
<span class="c1"># It is where you put the arguments to tophat that do not have &#39;flags&#39; to make</span>
<span class="c1"># their identity explicit like `-o path/to/output_dir` or `--library-type fr-unstranded`</span>

<span class="c1"># For tophat, these values are </span>
<span class="c1">#     [1] the bowtie index name</span>
<span class="c1">#     [2] the fastq files containing the left_reads</span>
<span class="c1">#     [3] the fastq files containing the right_reads</span>

<span class="c1"># They will be different for cufflinks, cuffmerge, cuffdiff so consult the</span>
<span class="c1"># respective help text or manuals, but you should be fine if you just use what</span>
<span class="c1"># I have set up in this file already.</span>

<span class="l-Scalar-Plain">tophat_options</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">library-type</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">fr-unstranded</span>
    <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">6</span>
    <span class="l-Scalar-Plain">r</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">125</span>
    <span class="l-Scalar-Plain">mate-std-dev</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">25</span>
    <span class="l-Scalar-Plain">G</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">no-coverage-search</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
    <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
        <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
        <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
        <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>

<span class="l-Scalar-Plain">cufflinks_options</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">7</span>
    <span class="l-Scalar-Plain">GTF-guide</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>  <span class="c1"># If you want to use annotation as *TRUTH* set this to False and set &#39;GTF&#39; to &#39;from_conditions&#39;</span>
    <span class="l-Scalar-Plain">GTF</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">False</span>                  <span class="c1"># if an option set to false, it will be ommited from the command string </span>
    <span class="l-Scalar-Plain">3-overhang-tolerance</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">5000</span>
    <span class="l-Scalar-Plain">frag-bias-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span> 
    <span class="l-Scalar-Plain">multi-read-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
    <span class="l-Scalar-Plain">upper-quartile-norm</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
    <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
        <span class="l-Scalar-Plain">accepted_hits</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>

<span class="l-Scalar-Plain">cuffmerge_options</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span> <span class="c1"># output directory</span>
    <span class="l-Scalar-Plain">ref-gtf</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">6</span>
    <span class="l-Scalar-Plain">ref-sequence</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
        <span class="l-Scalar-Plain">assembly_list</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span> <span class="c1"># file with path to cufflinks gtf files to be merged</span>

<span class="l-Scalar-Plain">cuffdiff_options</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">o</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">labels</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">p</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">6</span>
    <span class="l-Scalar-Plain">time-series</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
    <span class="l-Scalar-Plain">upper-quartile-norm</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
    <span class="l-Scalar-Plain">frag-bias-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">multi-read-correct</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">True</span>
    <span class="l-Scalar-Plain">positional_args</span><span class="p-Indicator">:</span>
        <span class="l-Scalar-Plain">transcripts_gtf</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
        <span class="l-Scalar-Plain">sample_bams</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>


<span class="l-Scalar-Plain">cummerbund_options</span><span class="p-Indicator">:</span>
    <span class="l-Scalar-Plain">cuffdiff-dir</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">gtf-path</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">out</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">from_conditions</span>
    <span class="l-Scalar-Plain">file-type</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">pdf</span>


<span class="c1"># options for --mode qsub_script</span>
<span class="c1"># If you are not using --mode qsub_script, then set all to &#39;None&#39;</span>
<span class="l-Scalar-Plain">qsub_options</span><span class="p-Indicator">:</span>
  <span class="l-Scalar-Plain">queues</span><span class="p-Indicator">:</span> <span class="s">&#39;queue1,queue3,queue5&#39;</span>
  <span class="l-Scalar-Plain">datahome</span><span class="p-Indicator">:</span> <span class="s">&#39;/path/to/baseDirectory/on/cluster/&#39;</span>
  <span class="l-Scalar-Plain">core_range</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">40-64</span> <span class="c1"># how many cpus do you want</span>
  <span class="l-Scalar-Plain">ld_library_path</span><span class="p-Indicator">:</span> <span class="s">&#39;&#39;</span>  <span class="c1"># leave this blank unless you know what it is and need it</span>
  <span class="l-Scalar-Plain">template</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/your/altered/version/of/qsub.template</span>


<span class="c1"># `condition_queue`:</span>
<span class="c1"># ------------------</span>
<span class="c1"># This is a list of info related to each sample/condition contained in your RNA-sequence</span>
<span class="c1"># experiment(s)</span>

<span class="c1"># `name`: the name of this condition program. Usually something like a time-point</span>
<span class="c1">#         ID or treatment type. Should be as short as possible while still being a useful label. </span>

<span class="c1"># `experiment_id`: this is how you group different experiments to be included in a</span>
<span class="c1">#             single cuffmerge/cuffdiff program call.  All conditions in a time</span>
<span class="c1">#             series should share the same `experiment_id` and be placed in</span>
<span class="c1">#             `condition_queue` in the order that you want them to be sent to</span>
<span class="c1">#             cuffdiff.</span>

<span class="c1"># `replicate_id`: this is how you group data for biological replicates of a single</span>
<span class="c1">#             experimental condition experiments to be included in a cuffdiff program</span>
<span class="c1">#             call.  Each replicate of a condition should have a unique `experiment_id`.</span>

<span class="c1"># `left_reads`: a list of the paths to fastq files containing left reads for</span>
<span class="c1">#               each condition. </span>

<span class="c1"># `right_reads`: list of fastqs containing the right mates for the fastqs in</span>
<span class="c1">#                `left_reads`.</span>
<span class="c1">#                 **NOTE** right mate file must be in same order as provided to `left_reads`</span>

<span class="l-Scalar-Plain">condition_queue</span><span class="p-Indicator">:</span>
    <span class="p-Indicator">-</span>
        <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp1_control</span>
        <span class="l-Scalar-Plain">experiment_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
        <span class="l-Scalar-Plain">replicate_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
        <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep1.left_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep2.left_reads.fastq</span>
        <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep1.right_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep2.right_reads.fastq</span>
        <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
        <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
        <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>

    <span class="p-Indicator">-</span>
        <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp1_control</span>
        <span class="l-Scalar-Plain">experiment_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
        <span class="l-Scalar-Plain">replicate_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">1</span>
        <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep1.left_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep2.left_reads.fastq</span>
        <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep1.right_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_control/techRep2.right_reads.fastq</span>
        <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
        <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
        <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>
        
    <span class="p-Indicator">-</span>
        <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp1_treatment</span>
        <span class="l-Scalar-Plain">experiment_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
        <span class="l-Scalar-Plain">replicate_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
        <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep1.left_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep2.left_reads.fastq</span>
        <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep1.right_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp1_treatment/techRep2.right_reads.fastq</span>
        <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
        <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
        <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>

    <span class="p-Indicator">-</span>
        <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp2_control</span>
        <span class="l-Scalar-Plain">experiment_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">1</span>
        <span class="l-Scalar-Plain">replicate_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
        <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep1.left_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep2.left_reads.fastq</span>
        <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep1.right_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_control/techRep2.right_reads.fastq</span>
        <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
        <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
        <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>

    <span class="p-Indicator">-</span>
        <span class="l-Scalar-Plain">name</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">exp2_treatment</span>
        <span class="l-Scalar-Plain">experiment_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">1</span>
        <span class="l-Scalar-Plain">replicate_id</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">0</span>
        <span class="l-Scalar-Plain">left_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep1.left_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep2.left_reads.fastq</span>
        <span class="l-Scalar-Plain">right_reads</span><span class="p-Indicator">:</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep1.right_reads.fastq</span>
            <span class="p-Indicator">-</span> <span class="l-Scalar-Plain">/path/to/exp2_treatment/techRep2.right_reads.fastq</span>
        <span class="l-Scalar-Plain">genome_seq</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/genome.fa</span>
        <span class="l-Scalar-Plain">gtf_annotation</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">/path/to/species/annotation.gtf</span>
        <span class="l-Scalar-Plain">bowtie2_index</span><span class="p-Indicator">:</span> <span class="l-Scalar-Plain">species.bowtie2_index.basename</span>


<span class="nn">...</span>
</pre></div>
</td></tr></table></div>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<p class="last">Add the slots for custom email server options.</p>
</div>
</div>
<div class="section" id="using-e-mail-notifications">
<h2>Using e-mail notifications<a class="headerlink" href="#using-e-mail-notifications" title="Permalink to this headline">¶</a></h2>
<div class="versionchanged">
<p><span>Changed in version v0.2.0rc1: </span>any smtp server should now be usable if you code the host and port into the yaml config file.  <em>Any</em> email can be used as the recipient.</p>
</div>
<div class="versionadded">
<p><span>New in version v0.2.0rc1: </span>added <tt class="docutils literal"><span class="pre">--no-email</span></tt> option.</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">gmail&#8217;s 2-step authentication will NOT work.  Sorry.  I will look into how to deal with that eventually.</p>
</div>
<p>You will need to provide your password in order to use the email notifications but it is not a good idea to store human readable passwords lying around your system.  So the file that is used to store your password must contain a version of your password that has been encoded in base64.  This will scramble your password beyond most people&#8217;s ability to read it as a password as long as you don&#8217;t name it something silly like <tt class="docutils literal"><span class="pre">password_file.txt</span></tt>.</p>
<p>The help text for <tt class="docutils literal"><span class="pre">blacktie-encode</span></tt> is:</p>
<div class="highlight-none"><div class="highlight"><pre>$ blacktie-encode -h

usage: blacktie-encode [-h] input_file

This script takes a path to a file where you have placed your password for the
email you want blacktie to use as the &quot;sender&quot; in its notification emails. It
will replace the file with one containing your password once it has encoded it
out of human readable plain-text into seemingly meaningless text. **THIS IS
NOT FOOLPROOF:** If someone knows exactly what to look for they might figure
it out. ALWAYS use good password practices and never use the same password for
multiple important accounts!

positional arguments:
    input_file  Path to a file where you have placed your password for the email
        you want blacktie to use as the &quot;sender&quot; in its notification
        emails.

optional arguments:
    -h, --help  show this help message and exit
</pre></div>
</div>
</div>
</div>
<div class="section" id="tutorial">
<h1>Tutorial<a class="headerlink" href="#tutorial" title="Permalink to this headline">¶</a></h1>
<p>A more detailed tutorial is under development, so <em>watch this space!</em></p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Getting started</a><ul>
<li><a class="reference internal" href="#the-prog-option">The &#8211;prog option</a></li>
<li><a class="reference internal" href="#the-hide-logs-option">The &#8211;hide-logs option</a></li>
<li><a class="reference internal" href="#the-modes-option">The &#8211;modes option</a></li>
<li><a class="reference internal" href="#the-configuration-file">The configuration file</a></li>
<li><a class="reference internal" href="#using-e-mail-notifications">Using e-mail notifications</a></li>
</ul>
</li>
<li><a class="reference internal" href="#tutorial">Tutorial</a></li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="install.html"
                        title="previous chapter">Installation</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="code.html"
                        title="next chapter">Blacktie Auto-Generated Code Documentation</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/tutorial.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="code.html" title="Blacktie Auto-Generated Code Documentation"
             >next</a> |</li>
        <li class="right" >
          <a href="install.html" title="Installation"
             >previous</a> |</li>
        <li><a href="index.html">Blacktie 0.2.1.2 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Augustine Dunn.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b1.
    </div>
  </body>
</html>